### DISCUSSION TITLE
Trying to understand basics of WinUI 3

### DISCUSSION THREAD

**Discussion by SDP190** [microsoft/microsoft-ui-xaml#9687] [General]
Are WinUI 3 controls a fullblown complete standalone project, or it relies on WinRT Windows.UI.Xaml.Controls?

Am looking for the code where Button, TextBox, CheckBox etc where are they implemented? cannot find anything in this repository.
So not all of WinUI 3 is in this repository? where can I find these basic core controls implementation?

As per the docs about WinUI 3:

> UX stack and control library completely decoupled from the OS and [Windows SDKs], including the core framework, composition, and input layers of the UX stack.

I do find the Xaml styles for these controls in dev/CommonStyles, but cannot find the actual type and functional implementation code anywhere in the repo.

**Comment by castorix** [comment]
> Are WinUI 3 controls a fullblown complete standalone project, or it relies on WinRT Windows.UI.Xaml.Controls?

Mainly on Microsoft.ui.xaml.dll
Sources are at (1.5) : https://github.com/microsoft/microsoft-ui-xaml/tree/winui3/release/1.5-stable
